luminance/src/tess.rs

```diff
@@ -393,24 +399,54 @@ pub enum TessError {
   Overflow(usize, usize)
 }
 
-pub struct Tess {
+pub unsafe trait VertIndex {
+  fn to_glenum() -> GLenum;
+  fn size() -> usize;
+}
+
+unsafe impl VertIndex for u8 {
+  fn to_glenum() -> GLenum {
+    gl::UNSIGNED_BYTE
+  }
+
+  fn size() -> usize {
+    size_of::<u8>()
+  }
+}
+
+unsafe impl VertIndex for u16 {
+  fn to_glenum() -> GLenum {
+    gl::UNSIGNED_SHORT
+  }
+
+  fn size() -> usize {
+    size_of::<u16>()
+  }
+}
+
+unsafe impl VertIndex for u32 {
+  fn to_glenum() -> GLenum {
+    gl::UNSIGNED_INT
+  }
+
+  fn size() -> usize {
+    size_of::<u32>()
+  }
+}
+
+pub struct Tess<I> {
```

Comment:
In the past (and current crate version), `Tess` had vertex type as phantom typing, so that not need is checked when working on vertices later (e.g. getting a slice mapped to memory to work on, etc.).

With deinterleaving and instancing, that tends to be harder to do and would also prevent dynamically set `Tess` (for instance, in a tool, when the vertex type is defined on-the-fly). Hence, I don’t really know why the type is needed here. Can you summarize your thoughts?

Reply:
The idea with using a type parameter here was to avoid a run-time check. If you want to keep `Tess` parameter-free, I can remove it easily.

Reply:
So it’s the same initial reason. I think it should be okay to keep it, as there’s no interleaving implied with indices.

Reply:
I had made it work with vertices as well a few weeks ago, but the public API is hard to use and won’t work with the correct builder-pattern `TessBuilder`. I think we should be able to merge that quickly.